src/algebraic_geometry/skyscraper.lean

```diff
@@ -0,0 +1,403 @@
+  (skyscraper_presheaf p₀ S ts).obj (op U) = S :=
+if_pos h
+
+lemma skyscraper_presheaf_obj_of_not_mem {U : opens X} (h : p₀ ∉ U) :
+  (skyscraper_presheaf p₀ S ts).obj (op U) = star :=
+if_neg h
+
+end
+
+/-
+A skyscraper sheaf is a sheaf supported at a single point: if `p₀ ∈ X` is a specified
+point, then the skyscraper sheaf `𝓕` with value `A` is defined by `U ↦ A` if `p₀ ∈ U` and
+`U ↦ *` if `p₀ ∉ A` where `*` is some terminal object.
+-/
+def skyscraper_sheaf : sheaf C X :=
+⟨skyscraper_presheaf p₀ S ts, λ c U s hs x hx,
+  ⟨dite (p₀ ∈ U)
+    (λ h, x (hs p₀ h).some_spec.some (hs p₀ h).some_spec.some_spec.1 ≫
+        eq_to_hom ((skyscraper_presheaf_obj_of_mem S ts (hs p₀ h).some_spec.some_spec.2).trans
+          (skyscraper_presheaf_obj_of_mem S ts h).symm))
+    (λ h, ts.from c ≫ (eq_to_hom (skyscraper_presheaf_obj_of_not_mem S ts h).symm)),
+    λ V inc h,
+    begin
+      by_cases hV : p₀ ∈ V,
+      { have hU : p₀ ∈ U := le_of_hom inc hV,
+        split_ifs,
+        generalize_proofs h₁ h₂ h₃ h₄,
+        dsimp,
+        split_ifs,
+        rw [category.id_comp, eq_to_hom_trans, category.assoc, eq_to_hom_trans],
+        generalize_proofs h₅,
+        have := hx _ _ h₂.some_spec.1 h rfl,
+        work_on_goal 2 { exact h₁.some ⊓ V },
+        work_on_goal 2 { exact hom_of_le inf_le_left },
+        work_on_goal 2 { exact hom_of_le inf_le_right },
+        dsimp at this,
+        have hV' : p₀ ∈ h₁.some ⊓ V := ⟨h₂.some_spec.2, hV⟩,
+        split_ifs at this,
+        rw [category.id_comp, eq_to_hom_trans, eq_to_hom_trans, eq_comp_eq_to_hom,
+          category.assoc, eq_to_hom_trans] at this,
+        generalize_proofs h₆ at this,
+        rw [this, eq_comp_eq_to_hom], },
+      { dsimp,
+        split_ifs,
+        rw [←category.assoc, eq_comp_eq_to_hom],
+        exact ts.hom_ext _ _, }
+    end,
+    λ y (hy : x.is_amalgamation y),
+    begin
+      split_ifs,
+      { generalize_proofs h₁ h₂ h₃ h₄,
+        have := hy h₂.some h₂.some_spec.1,
+        dsimp at this,
+        split_ifs at this with H,
+        { rw [category.id_comp, eq_to_hom_trans, eq_comp_eq_to_hom] at this,
+          rw [this, eq_comp_eq_to_hom, category.assoc, eq_to_hom_trans, eq_to_hom_refl,
+            category.comp_id], },
+        { exfalso,
+          exact H h₂.some_spec.2, }, },
+      { rw [←eq_comp_eq_to_hom],
+        exact (ts.hom_ext _ _), }
+    end⟩⟩
+
+example : true := trivial
+
+end
+
+section
+
+-- In this section, we calculate the stalks for skyscraper presheaves. We need to restrict universe
+-- level.
+
+open topological_space
+open category_theory category_theory.limits
+open Top
+open opposite
+
+universes u v
+
+variables {X : Top.{u}} (p₀ : X) {C : Type v} [category.{u} C] (S : C)
+variables {star : C} (ts : is_terminal star)
+variable [Π (U : opens X), decidable (p₀ ∈ U)]
+variable [has_colimits C]
+
+lemma mem_nhds_of_closure_singleton {y : X} (U : open_nhds y) (h : y ∈ (closure ({p₀} : set X))) :
```

Comment:
You can spell `h` as `p₀ ⤳ y` ([docs#specializes](https://leanprover-community.github.io/mathlib_docs/topology/inseparable.html#specializes)), and this lemma (and the following one) should follow from some general filter magic.

Reply:
done